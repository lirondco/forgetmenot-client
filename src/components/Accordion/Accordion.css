@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600&display=swap");

/* Style the accordion section */
.accordion__section {
  display: flex;
  flex-direction: column;
}

/* Style the buttons that are used to open and close the accordion panel */
.accordion {
  background-color: #C03D72;
  color: #EFEBE9;
  cursor: pointer;
  padding: 1%;
  height: 50px !important;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 14px;
  outline: none;
  transition: background-color 0.6s ease;
  margin: 1%;
}

/* Add a background color to the button if it is clicked on (add the .active class with JS), and when you move the mouse over it (hover) */
.accordion:hover,
.active {
  background-color: #C03D72;
}

/* Style the accordion content title */
.accordion__title {
  font-family: 'Patrick Hand SC', cursive;
  font-weight: 600;
  font-size: 27px;
  text-align: left;
}

/* Style the accordion chevron icon */
.accordion__icon {
  margin-left: auto;
  transition: transform 0.6s ease;
}

/* Style to rotate icon when state is active */
.rotate {
  transform: rotate(90deg);
}

/* Style the accordion content panel. Note: hidden by default */
.accordion__content {
  background-color: #9B898B;
  color: #EFEBE9;
  overflow: auto;
  transition: max-height 0.6s ease;
  border-radius: 14px;
  margin: 1%;
}

/* Style the accordion content text */
.accordion__text {
  font-family: 'Mina', sans-serif;
  font-weight: 400;
  font-size: 18px;
  padding: 18px;
}

/* Style of delete button */
.delete_idea {
  order: 1px solid #aaa;
  background-color: #FF6F00;
  border-radius: 4px;
  margin: 0 0 5% 35%;
  padding: 1%;
  font-family: 'Mina', sans-serif;
  text-align: center;
}