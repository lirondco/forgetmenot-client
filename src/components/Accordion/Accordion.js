import React, { useState, useRef } from "react";
import Chevron from "./Chevron";

import "./Accordion.css";
import IdeasApiService from "../../services/ideas-api-service";

/* this is an 'accordion' component that I used to house each of the user's
 idea. c/o with some personalised modifications:
 https://medium.com/skillthrive/build-a-react-accordion-component-from-scratch-using-react-hooks-a71d3d91324b */

function Accordion(props) {
  const [setActive, setActiveState] = useState("");
  const [setHeight, setHeightState] = useState("0px");
  const [setRotate, setRotateState] = useState("accordion__icon");

  const content = useRef(null);

  function toggleAccordion() {
    setActiveState(setActive === "" ? "active" : "");
    setHeightState(
      setActive === "active" ? "0px" : `${content.current.scrollHeight}px`
    );
    setRotateState(
      setActive === "active" ? "accordion__icon" : "accordion__icon rotate"
    );
  }

  const handleClickDelete = e => {
    e.preventDefault()
    const ideaId = props.id
    IdeasApiService.deleteIdea(ideaId)
     .then(() => props.onDeleteSuccess())
     .catch(error=>console.error({error}))
  }

  return (
    <div className="accordion__section">
      <div className={`accordion ${setActive}`} onClick={toggleAccordion}>
        <p className="accordion__title">{props.title}</p>
        <Chevron className={`${setRotate}`} width={10} fill={"#EFEBE9"} />
      </div>
      <div
        ref={content}
        style={{ maxHeight: `${setHeight}` }}
        className="accordion__content"
      >
        <div
          className="accordion__text"
          dangerouslySetInnerHTML={{ __html: props.content }}
        />
        <button type='button' className='delete_idea' onClick={handleClickDelete}>
            Delete Idea {/* added this button for idea deletion function */}
        </button>
      </div>
    </div>
  );
}

export default Accordion;
