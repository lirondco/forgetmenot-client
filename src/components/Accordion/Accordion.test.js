import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe(`List component`, () => {
  const props = {
    list: {
        title: "accordion title",
        content: "accordion content"
    }
  }

  it('renders an accordion by default', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the accordion given props', () => {
    const wrapper = shallow(<Accordion {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('button works when clicked', () => {
    const wrapper = shallow(<Accordion {...props} />)
    wrapper.find('button').simulate('click', { preventDefault() {}})
})

})