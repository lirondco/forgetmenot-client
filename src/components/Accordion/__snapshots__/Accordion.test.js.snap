// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders an accordion by default 1`] = `
<div
  className="accordion__section"
>
  <div
    className="accordion "
    onClick={[Function]}
  >
    <p
      className="accordion__title"
    />
    <Chevron
      className="accordion__icon"
      fill="#EFEBE9"
      width={10}
    />
  </div>
  <div
    className="accordion__content"
    style={
      Object {
        "maxHeight": "0px",
      }
    }
  >
    <div
      className="accordion__text"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <button
      className="delete_idea"
      onClick={[Function]}
      type="button"
    >
      Delete Idea 
    </button>
  </div>
</div>
`;

exports[`List component renders the accordion given props 1`] = `
<div
  className="accordion__section"
>
  <div
    className="accordion "
    onClick={[Function]}
  >
    <p
      className="accordion__title"
    />
    <Chevron
      className="accordion__icon"
      fill="#EFEBE9"
      width={10}
    />
  </div>
  <div
    className="accordion__content"
    style={
      Object {
        "maxHeight": "0px",
      }
    }
  >
    <div
      className="accordion__text"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <button
      className="delete_idea"
      onClick={[Function]}
      type="button"
    >
      Delete Idea 
    </button>
  </div>
</div>
`;
